pkg/db/db.go

```diff
@@ -247,3 +250,18 @@ func deserializeRecord(r *bufio.Reader) (key string, val []byte, seq uint64, err
 	}
 	return
 }
+
+func packSeq(seq uint64, repro bool) uint64 {
+	if repro {
```

Comment:
Would be useful to check that seq does not have high bit set. Some caller can use e.g. -1 as a special value.
In fact, seqDeleted has high bit set, and such check would help you find the bug in deserializeRecord.